api/src/main/java/org/apache/iceberg/types/Conversions.java

```diff
@@ -117,7 +117,12 @@ public static ByteBuffer toByteBuffer(Type type, Object value) {
   }
 
   private static Object internalFromByteBuffer(Type type, ByteBuffer buffer) {
-    ByteBuffer tmp = buffer.duplicate().order(ByteOrder.LITTLE_ENDIAN);
+    ByteBuffer tmp = buffer.duplicate();
+    if (type == Types.UUIDType.get() || type instanceof Types.DecimalType) {
```

Comment:
I set the byte order for decimals explicitly to avoid any confusion. However, it is not mandatory because of the way we read bytes later.

```
case DECIMAL:
  Types.DecimalType decimal = (Types.DecimalType) type;
  byte[] unscaledBytes = new byte[buffer.remaining()];
  tmp.get(unscaledBytes);
  return new BigDecimal(new BigInteger(unscaledBytes), decimal.scale());
```

The byte order doesn't matter in the code snippet above.